tests/worlds/mazeworld/test_weather.py

```diff
@@ -0,0 +1,132 @@
+        {
+            "agent_name": "sphere0",
+            "agent_type": "SphereAgent",
+            "sensors": [
+                {
+                    "sensor_type": "RGBCamera",
+                    "socket": "CameraSocket",
+                    "sensor_name": "TestCamera"
+                }
+            ],
+            "control_scheme": 0,
+            "location": [.95, -1.75, .5]
+        }
+    ],
+    "window_width": 1024,
+    "window_height": 1024
+}
+
+
+def env_with_config(config):
+    binary_path = pm.get_binary_path_for_package("DefaultWorlds")
+    return HolodeckEnvironment(scenario=config,
+                               binary_path=binary_path,
+                               show_viewport=False,
+                               uuid=str(uuid.uuid4()))
+
+
+def cv2_show_images(images: List[Tuple[str, List]]):
+    """
+    Show list of images in OpenCV image windows when debugging
+    """
+    for image in images:
+        cv2.imshow(image[0], image[1])
+    cv2.waitKey(0)
+    cv2.destroyAllWindows()
+
+
+def compare_rgb_with_baseline(sensor_data, base_path, baseline_name, show_images=False) -> float:
+    """
+    Compare data from RGB sensor with baseline file in `expected` folder and return mean squared error
+    """
+    pixels = sensor_data[:, :, 0:3]
+    baseline = cv2.imread(os.path.join(base_path, "expected", f"{baseline_name}.png"))
+    if show_images:
+        # Show images when debugging--this will block tests until user input is provided
+        display_multiple([(pixels, "pixels"), (baseline, "baseline")])
+    return mean_square_err(pixels, baseline)
+
+
+@pytest.mark.parametrize("weather_type, max_err", weather_type_test_data)
+def test_weather_type_scenario(weather_type: str, max_err: float, request: FixtureRequest) -> None:
+    current_config = config.copy()
+    current_config["weather"] = {"type": weather_type}
+
+    with env_with_config(current_config) as env:
+        for _ in range(5):
+            env.tick()
+        err = compare_rgb_with_baseline(env.tick()["TestCamera"], request.fspath.dirname,
+                                        f"weather_type_{weather_type}")
+        assert err < max_err
+
+
+@pytest.mark.parametrize("fog_density, max_err", fog_test_data)
+def test_weather_fog_scenario(fog_density: float, max_err: float, request: FixtureRequest) -> None:
+    current_config = config.copy()
+    current_config["weather"] = {"fog_density": fog_density}
+
+    with env_with_config(current_config) as env:
+        for _ in range(5):
+            env.tick()
+
+        err = compare_rgb_with_baseline(env.tick()["TestCamera"], request.fspath.dirname,
+                                        f"weather_fog_density_{fog_density}")
+        assert err < max_err
+
+
+@pytest.mark.parametrize("weather_type, max_err", weather_type_test_data)
+def test_weather_type_programmatic(weather_type: str, max_err: float, request: FixtureRequest) -> None:
+    with env_with_config(config) as env:
```

Comment:
Since the other tests were validating that it parses the config correctly, it is OK that you made a new instance environment every time. But in this test you could share an environment between tests and shave off ~10s of waiting for the engine to load. 

The way you've done it here it isn't horrible since it only makes 3 environments total, but generally if we can get away with it, an environment should be shared between different instances of a parameterized test, or even between tests. Otherwise running them is going to balloon more than it has (I had a job where running tests took ~10 hours so I'm scarred haha)

I've previously done this in `conftest.py`. Here you don't have to, but I wanted you to be aware of this.

If you make a test that validates the weather in every scenario, we already load up every available scenario in `tests/scenarios/conftest.py`, and you absolutely should use those instances.